commons-vfs2/src/main/java/org/apache/commons/vfs2/FileExtensionSelector.java

```diff
@@ -70,15 +72,7 @@ public FileExtensionSelector(final String... extensions) {
      */
     @Override
     public boolean includeFile(final FileSelectInfo fileInfo) throws Exception {
-        if (this.extensions == null) {
-            return false;
-        }
-        for (final String extension : this.extensions) {
-            if (fileInfo.getFile().getName().getExtension().equalsIgnoreCase(extension)) {
-                return true;
-            }
-        }
-        return false;
+        return this.extensions.contains(fileInfo.getFile().getName().getExtension().toLowerCase());

```

Comment:
I'm not sure that's quite right. Unit tests would help check that case is handled properly. Note that not all file systems are case-insenstive. OTOH, if that is an issue with this code, it was already there! ;-)